src/irmin-pack/unix/gc.ml

```diff
@@ -322,7 +325,8 @@ module Worker = struct
             let off = Int63.Syntax.(off + len) in
             transfer_loop ~off (i - 1)
       in
-      let offs = transfer_loop ~off:commit_offset num_iterations in
+      let after_commit_offset = Int63.Syntax.(commit_offset + commit_length) in
```

Comment:
It's a shame to have this offset calculation duplicated here and in `ext`. I don't think we should do anything in this PR (it simply highlights the spread of this logic) but it would be nice to have this offset calculation in one place (one idea is to move some of the logic from `ext` into `Gc.v` and then pass this offset to the worker).

Reply:
I added a commit with this change: we now pass both `commit_key` and `new_prefix_end_offset` to the worker there is a bit of redundancy as the `commit_key` is sufficient to compute the `after_commit_offset` from it. But both work for me.

Reply:
Cool, thanks. I like this change since the name also helps to describe intent. There are a few places of duplicated calculations/work related to the `commit_key` that maybe we can clean up at some point. :smile: